### PR TITLE
Add projectId config option to GCS techdocs publisher

### LINKED ISSUE #13049: Allow explicitly setting project ID in techdocs GCS publisher
## Feature Suggestion

Add a `projectId` field to the `googleGcs` publisher config object, and use that to initialise the storage client if present.

## Possible Implementation

```yaml
techdocs:
  publisher:
    type: 'googleGcs'
    googleGcs:
      bucketName: 'my-bucket-1'
      credentials: $GOOGLE_CREDENTIALS
      projectId: 'my-project-1'
```

Then, when initialising the storage client, you would set the `projectId` from the config.

## Context

Currently the publisher grabs the project ID willy-nilly from the credentials blob it uses. CF 
https://github.com/backstage/backstage/blob/master/plugins/techdocs-node/src/stages/publish/googleStorage.ts#L97-L102

If no credentials are provided, no arguments are passed to the storage client.

This leads to a problem in cases where:
1. the credentials provided are for a service account in a different project to the bucket
2. the credentials are end user credentials, which don't have a project associated with them.

Backstage will not be able to read the docs (so to speak 🥁 ) in these cases.

In these cases, it would be better to be able to just explicitly set the project with the bucket in it.

### PR DESCRIPTION
**Pull request by tragiclifestories**
Fixes #13049

## Hey, I just made a Pull Request!

Add `projectId` config option to GCP Cloud Storage techdocs publisher. This will allow users to override the project ID, instead of implicitly using the same one as found in a credentials bundle.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by tragiclifestories** [comment]
I haven't added a test for this because `fromConfig` is not yet undefined, and I can't see how you'd test this without breaking changes to the API (that would allow passing some kind of mockable `StorageClientFactory` to it).

File: docs/features/techdocs/using-cloud-storage.md
Comment (soapraj) [inline docs/features/techdocs/using-cloud-storage.md:41]: Maybe `techdocs.publisher.googleGcs.projectId` needs a separate comment ?

Comment (tragiclifestories) [reply docs/features/techdocs/using-cloud-storage.md]: Good point - I've rephrased this.

**Review by emmaindal — APPROVED** [review]
Looks good to me! Thanks for the contribution. I'll let @realandersn have a final look and merge if no other comments are added.

File: docs/features/techdocs/using-cloud-storage.md
Comment (realandersn) [inline docs/features/techdocs/using-cloud-storage.md:41]: Not sure this is the place to document it. Reading these makes me think setting it is required when it is in fact optional.

File: docs/features/techdocs/using-cloud-storage.md
Comment (realandersn) [inline docs/features/techdocs/using-cloud-storage.md:104]: Maybe turning it around, stating when you need it rather than when you don't. Something like: "If the service account used is not in the same project as the bucket, you need to set the `projectId` field."

**Review by realandersn — COMMENTED** [review]
Just a few documentation comments. I think we need @backstage/reviewers approval as well.

File: docs/features/techdocs/using-cloud-storage.md
Comment (realandersn) [reply docs/features/techdocs/using-cloud-storage.md:41]: Alright, let's get it in as is and we can iterate if needed.

**Review by realandersn — APPROVED** [review]

File: .changeset/gold-laws-attend.md
Comment (Rugvip) [inline .changeset/gold-laws-attend.md:6]:
```suggestion
Add `projectId` config option to GCP Cloud Storage techdocs publisher. This will allow users to override the project ID, instead of implicitly using the same one as found in a credentials bundle.
```

File: .github/vale/Vocab/Backstage/accept.txt
Comment (Rugvip) [inline .github/vale/Vocab/Backstage/accept.txt:238]: Let's get rid of this here and use in-line code blocks wherever it's used in docs.

Comment (tragiclifestories) [reply .github/vale/Vocab/Backstage/accept.txt]: TIL ...

**Pull request merged**
